src/client/AppEmbeddedSDK.ts

```diff
@@ -0,0 +1,57 @@
+import { ACTION_ID_LENGTH } from './constants';
+import { randomString } from './utils';
+
+export class AppEmbeddedSDK {
+    private listener: (this: Window, ev: MessageEvent) => any;
+    private callbacks: Map<string, (response: any) => any>;
+
+    constructor() {
+        this.listener = () => console.log('init');
+        this.callbacks = new Map();
+    }
+
+    public getUserInfo(): Promise<any> {
+        return this.call(AppEmbeddedSDKActions.GET_USER_INFO);
+    }
+
+    public getRoomInfo(): Promise<any> {
```

Comment:
`Promise<any>`? Also, where is the documentation for this?

Reply:
Hi @graywolf336 . Here is the temporal documentation for the AppEmbeddedSDK: https://lolimay.gitbook.io/workspace. Can you please also review it? It will be great if you can also leave some comments :)
I'll create a PR to add this document to the docs repository if there's no much problem.

Reply:
It looks okay for external documentation, however, our current codebase has some documentation generated here: https://rocketchat.github.io/Rocket.Chat.Apps-engine/. Thus it would be not only beneficial but also consistent to have documentation inside the codebase on the code. Another thing, please remove the `any` and have it be typed so that developers know exactly what to expect and they get code suggestions in their IDE which is one of the biggest benefits for using TypeScript.

Reply:
Got it. Thanks! I will do it.

Reply:
Also, I see that in your documentation people have to link to a CDN for the file. We have a large percentage of Rocket.Chat servers who operate offline and with their Rocket.Chat instance not being able to talk to the internet. How will external components work in this instance? We are working on enabling "air gapped" servers to download and install Rocket.Chat Apps from the Marketplace, which means Apps will need to be able to operate fully offline. How will external components work with these type systems?

Reply:
It's a temporal solution that the FRVR team can use our AppEmbeddedSDK for testing at an early stage. Later, we will release compiled SDK files on the npm. So that the developers can `npm install` to install the AppEmbeddedSDK and buddle it with their apps together.

And I don't think the external components can be loaded in an offline environment. Because the HTML5 game is usually served on a game company server or a CDN. This means the "air-gapped" hoster should give up the external component feature that can play chat-based HTML5 games in Rocket.Chat.

Reply:
@geekgonecrazy I made a [PR](https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/147) to add types for the public AppEmbeddedSDK methods and add documentations for them. Can you please review it?

Reply:
Okay. So, it sounds like we might need to add a flag for app publishers to tell us which Apps support "air gapped" servers and which ones require an internet connection. @d-gubert and @geekgonecrazy would like your opinions and thoughts on this.

Reply:
I agree I think would be worth while to identify all apps that must have an internet connection to work.  I think this would even be useful for displaying to admin's.

Reply:
I agree, and that may be a permission in the future as well.

Reply:
As of right now we could assume that. If they use `IHttp` or implement this interface, they require an internet connection to work.

Reply:
An app that needs to register an external component also possibly requires an Internet connection to work (The external component usually creates an `<iframe>` container to load HTML5 game by the external URL).

Reply:
Yes - but in a different moment. The "airgapped" we usually refer is regarding the servers, so if your app wants to make a request to the internet in a slashcommand, for instance, it wouldn't be possible.

But as the `<iframe>` is rendered on the UI, the user would more likely have access to the outer world - but maybe restricted by some firewall rules